core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -386,8 +386,10 @@ class LogSegment(val log: FileRecords,
    */
   def changeFileSuffixes(oldSuffix: String, newSuffix: String) {
 
-    def kafkaStorageException(fileType: String, e: IOException) =
+    def kafkaStorageException(fileType: String, e: IOException) = {
```

Comment:
Good point. I have updated the code to the following with an extra comment. I think we don't have to catch IOException for logging purpose here because the IOException should specify the file name and be caught and logged by the caller.

```
  /**
   * Change the suffix for the index and log file for this log segment
   * IOException from this method should be logged and handled by the caller
   */
  def changeFileSuffixes(oldSuffix: String, newSuffix: String) {
    log.renameTo(new File(CoreUtils.replaceSuffix(log.file.getPath, oldSuffix, newSuffix)))
    index.renameTo(new File(CoreUtils.replaceSuffix(index.file.getPath, oldSuffix, newSuffix)))
    timeIndex.renameTo(new File(CoreUtils.replaceSuffix(timeIndex.file.getPath, oldSuffix, newSuffix)))
    txnIndex.renameTo(new File(CoreUtils.replaceSuffix(txnIndex.file.getPath, oldSuffix, newSuffix)))
  }
```